### ISSUE TITLE
`switch-to-buffer` is broken on master

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2370]

**Comment by Ambrevar** [comment]
Please share a backtrace at least.  Also it's probably the same bug as #2369, so 1 issue suffices here ;)

**Comment by aadcg** [comment]
```
The value
  #<NYXT:WEB-BUFFER 1888 {1007B48833}>
is not of type
  LIST
   [Condition of type TYPE-ERROR]

Restarts:
 0: [ABORT] abort thread (#<THREAD "Nyxt run-async command" RUNNING {10074BAE23}>)

Backtrace:
 0: (NYXT:PROMPT1 :PROMPT "Switch to buffer" :SOURCES (#<NYXT:BUFFER-SOURCE {10083954E3}>))
      Locals:
        SB-DEBUG::MORE = (:PROMPT "Switch to buffer" :SOURCES (#<NYXT:BUFFER-SOURCE {10083954E3}>))
 1: ((:METHOD NYXT:SWITCH-BUFFER NIL) :CURRENT-IS-LAST-P NIL :BUFFER NIL) [fast-method]
      Locals:
        #:.DEFAULTING-TEMP. = NIL
        NYXT::CURRENT-IS-LAST-P = NIL
        #:N-SUPPLIED-3 = 0
 2: (NYXT::RUN-COMMAND #<NYXT:COMMAND NYXT:SWITCH-BUFFER (1)> NIL)
 3: ((LAMBDA NIL :IN NYXT::RUN-ASYNC))
 4: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
 5: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
 6: ((FLET "WITHOUT-INTERRUPTS-BODY-10" :IN SB-THREAD::RUN))
 7: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
 8: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-THREAD::RUN))
 9: (SB-THREAD::RUN)
10: ("foreign function: call_into_lisp")
11: ("foreign function: funcall1")
 --more--
```

Sorry for not adding the backtrace immediately.

**Comment by Ambrevar** [comment]
Thanks, should be fixed with 3e7682f81c9c4f4f38a7bb8ea25e3c02d65f5bde.

**Comment by Ambrevar** [comment]
To clarify, switch-buffer did work, it's just that a condition was raised (but only when run from the REPL), upon returning.

Feel free to reopen if there is anything else.
